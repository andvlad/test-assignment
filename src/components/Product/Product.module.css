.root {
  max-width: 320px;
  margin: 0 25px 25px 0;
  padding: 12px;
  border-radius: 8px;
  -webkit-box-shadow: 0px 0px 10px 2px rgba(34, 60, 80, 0.2);
  -moz-box-shadow: 0px 0px 10px 2px rgba(34, 60, 80, 0.2);
  box-shadow: 0px 0px 10px 2px rgba(34, 60, 80, 0.2);
}

.title {
  font-size: 18px;
  font-weight: 600;
  margin-bottom: 8px;
}

.text {
  font-size: 16px;
}

.highlighted {
  background-color: yellow;
}
