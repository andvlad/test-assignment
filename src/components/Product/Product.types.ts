export interface IProduct {
  isInternal: boolean;
  name: string;
  plu: string;
  price: number;
  productType: number;
  quantity: number;
  sortOrder: number;
  subItems: IProduct[];
}
