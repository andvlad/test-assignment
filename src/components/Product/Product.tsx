import React from 'react';

import { IProduct } from './Product.types';
import styles from './Product.module.css';

export function Product({
  name,
  price,
  quantity,
}: IProduct) {
  return (
    <div className={styles.root}>
      <div className={styles.title}>
        {name}
      </div>
      <div className={styles.quantity}>
        Quantity: {quantity}
      </div>
      <div className={styles.price}>
        Price: {price}
      </div>
    </div>
  );
}
