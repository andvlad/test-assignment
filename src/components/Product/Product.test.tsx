import React from 'react';
import { render } from '@testing-library/react';

import { Product } from './Product';

const productData = {
  isInternal: true,
  name: 'Hawaii Burger',
  plu: 'CAT1-0001',
  price: 0,
  productType: 1,
  quantity: 1,
  sortOrder: 0,
  subItems: [],
};

it('matches snapshot', () => {
  const { asFragment } = render(
    <Product {...productData} />
  );

  expect(asFragment()).toMatchSnapshot();
});
