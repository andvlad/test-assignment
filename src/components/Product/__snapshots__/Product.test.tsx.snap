// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="title"
    >
      Hawaii Burger
    </div>
    <div
      class="quantity"
    >
      Quantity: 1
    </div>
    <div
      class="price"
    >
      Price: 0
    </div>
  </div>
</DocumentFragment>
`;
